tlslite/tlsconnection.py

```diff
@@ -782,6 +834,196 @@ def _clientGetServerHello(self, settings, clientHello):
+        self._recordLayer.calcTLS1_3PendingState(
+            serverHello.cipher_suite,
+            cl_handshake_traffic_secret,
+            sr_handshake_traffic_secret,
+            settings.cipherImplementations)
+
+        self._changeReadState()
+
+        for result in self._getMsg(ContentType.handshake,
+                                   HandshakeType.encrypted_extensions):
+            if result in (0, 1):
+                yield result
+            else:
+                break
+        encrypted_extensions = result
+        assert isinstance(encrypted_extensions, EncryptedExtensions)
+
+        for result in self._getMsg(ContentType.handshake,
+                                   HandshakeType.certificate,
+                                   CertificateType.x509):
+            if result in (0, 1):
+                yield result
+            else:
+                break
+
+        certificate = result
+        assert isinstance(certificate, Certificate)
+
+        srv_cert_verify_hh = self._handshake_hash.copy()
+
+        for result in self._getMsg(ContentType.handshake,
+                                   HandshakeType.certificate_verify):
+            if result in (0, 1):
+                yield result
+            else:
+                break
+        certificate_verify = result
+        assert isinstance(certificate_verify, CertificateVerify)
+
+        signature_scheme = certificate_verify.signatureAlgorithm
+
+        scheme = SignatureScheme.toRepr(signature_scheme)
+        # keyType = SignatureScheme.getKeyType(scheme)
+        padType = SignatureScheme.getPadding(scheme)
+        hashName = SignatureScheme.getHash(scheme)
+        saltLen = getattr(hashlib, hashName)().digest_size
+
+        signature_context = bytearray(b'\x20' * 64 +
+                                      b'TLS 1.3, server CertificateVerify' +
+                                      b'\x00') + \
+                            srv_cert_verify_hh.digest(prfName)
+
+        signature_context = secureHash(signature_context, hashName)
+
+        publicKey = certificate.certChain.getEndEntityPublicKey()
+
+        if not publicKey.verify(certificate_verify.signature,
+                                signature_context,
+                                padType,
+                                hashName,
+                                saltLen):
+            raise TLSDecryptionFailed("server Certificate Verify signature "
+                                      "verification failed")
+
+        transcript_hash = self._handshake_hash.digest(prfName)
+
+        for result in self._getMsg(ContentType.handshake,
+                                   HandshakeType.finished,
+                                   prf_size):
+            if result in (0, 1):
+                yield result
+            else:
+                break
+        finished = result
+
+        server_finish_hs = self._handshake_hash.copy()
+
+        assert isinstance(finished, Finished)
```

Comment:
nit: shouldn't this assert be right after `finished = result` ?

Reply:
`_handshake_hash.copy()` will never fail so there's little difference in placement (and it's just an assert, I expect it to be removed by optimised builds)